internal/service/project/resource_project_test.go

```diff
@@ -1,20 +1,441 @@
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/project"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/stretchr/testify/mock"
 	"go.mongodb.org/atlas-sdk/v20231115002/admin"
-	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
+var (
+	successfulPaginatedTeamRole = ProjectResponse{
+		ProjectTeamResp: &admin.PaginatedTeamRole{},
+		Err:             nil,
+	}
+	successfulDataFederationLimit = ProjectResponse{
+		LimitsResponse: []admin.DataFederationLimit{},
+		Err:            nil,
+	}
+	successfulGroupSettingsResponse = ProjectResponse{
+		GroupSettingsResponse: &admin.GroupSettings{},
+		Err:                   nil,
+	}
+	name             = types.StringValue("sameName")
+	diffName         = types.StringValue("diffName")
+	projectStateName = project.TfProjectRSModel{
+		Name: name,
+	}
+	projectStateNameDiff = project.TfProjectRSModel{
+		Name: diffName,
+	}
+)
+
+func TestGetProjectPropsFromAPI(t *testing.T) {
+	testCases := []struct {
+		name          string
+		projectID     string
+		mockResponses []ProjectResponse
+		expectedError bool
+	}{
+		{
+			name: "Successful",
+			mockResponses: []ProjectResponse{
+				successfulPaginatedTeamRole,
+				successfulDataFederationLimit,
+				successfulGroupSettingsResponse,
+			},
+			expectedError: false,
+			projectID:     "projectId",
+		},
+		{
+			name: "Fail to get project's teams assigned ",
+			mockResponses: []ProjectResponse{
+				{
+					ProjectTeamResp: nil,
+					HTTPResponse:    &http.Response{StatusCode: 503},
+					Err:             errors.New("Service Unavailable"),
+				},
+			},
+			expectedError: true,
+			projectID:     "projectId",
+		},
+		{
+			name: "Fail to get project's limits",
+			mockResponses: []ProjectResponse{
+				successfulPaginatedTeamRole,
+				{
+					LimitsResponse: nil,
+					HTTPResponse:   &http.Response{StatusCode: 503},
+					Err:            errors.New("Service Unavailable"),
+				},
+			},
+			expectedError: true,
+			projectID:     "projectId",
+		},
+		{
+			name: "Fail to get project's settings",
+			mockResponses: []ProjectResponse{
+				successfulPaginatedTeamRole,
+				successfulDataFederationLimit,
+				{
+					GroupSettingsResponse: nil,
+					HTTPResponse:          &http.Response{StatusCode: 503},
+					Err:                   errors.New("Service Unavailable"),
+				},
+			},
+			expectedError: true,
+			projectID:     "projectId",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testObject := new(MockProjectService)
+			if len(tc.mockResponses) > 0 {
+				testObject.On("ListProjectTeams", mock.Anything, mock.Anything).Return(tc.mockResponses[0])
+			}
+			if len(tc.mockResponses) > 1 {
+				testObject.On("ListProjectLimits", mock.Anything, mock.Anything).Return(tc.mockResponses[1])
+			}
+			if len(tc.mockResponses) > 2 {
+				testObject.On("GetProjectSettings", mock.Anything, mock.Anything).Return(tc.mockResponses[2])
+			}
```

Comment:
nit: for readability, might find it more clear to have the 3 mocked responses in separate fields defined in the test fixture, and have the 3 calls mocked for all test cases.